### PR TITLE
feat(user): add agyn_user resource

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- add UsersGateway client support and CRUD agyn_user resource with cluster role handling
- include users protos in buf generation and add docs for the new resource

## Testing
- buf dep update
- buf generate
- go generate ./... (tools)
- go test ./...

## Issue
- #362

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Summary
- add UsersGateway client support and CRUD agyn_user resource with cluster role handling
- generate docs for the new resource via go generate

## Test & Lint Summary
- buf dep update
- buf generate
- go generate ./... (tools)
- go test ./...

File: internal/resources/user_resource.go [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+		needsUpdate = true
+	}
+	if input.Nickname != nil {
+		updateReq.Nickname = input.Nickname
+		needsUpdate = true
+	}
+	if clusterRole := clusterRolePointerFromPlan(plan.ClusterRole); clusterRole != nil {
+		updateReq.ClusterRole = clusterRole
+		needsUpdate = true
+	}
```
Comment (noa-lucent) [inline internal/resources/user_resource.go]: [minor] The `Name`, `PhotoUrl`, and `Nickname` fields are already sent in the `CreateUserRequest` (lines 103-105), then re-sent identically in the `UpdateUserRequest` (lines 121-131). Only `ClusterRole` actually needs the follow-up call.

Consider gating the update solely on `ClusterRole`:

```go
if clusterRole := clusterRolePointerFromPlan(plan.ClusterRole); clusterRole != nil {
    updateReq := &usersv1.UpdateUserRequest{
        IdentityId:  identityID,
        ClusterRole: clusterRole,
    }
    if _, err := r.client.UpdateUser(ctx, updateReq); err != nil {
        resp.Diagnostics.AddError("Unable to update user cluster role", err.Error())
        return
    }
}
```

This removes the redundant field writes and makes the intent clearer: "create the user, then optionally promote to admin."

File: internal/resources/user_resource.go [hunk 2]
```diff
@@ -0,0 +1,335 @@
+package resources
```
Comment (noa-lucent) [inline internal/resources/user_resource.go:1]: [major] Every other resource in this provider has acceptance tests (`internal/provider/resource_*_test.go`). This new resource is missing a corresponding `resource_user_test.go`.

Please add acceptance tests covering at minimum:
- **basic create** — create a user with `oidc_subject` and verify `identity_id` is set
- **update** — change `cluster_role` (e.g. `none` → `admin`) and verify
- **import** — import by `identity_id` and verify state

The existing `resource_organization_test.go` is a good template to follow.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Clean implementation that follows the established provider patterns well. The client layer, schema, CRUD operations, role conversion, and docs are all solid.

Two items to address:

1. **[major] Missing acceptance tests.** Every other resource has test coverage — this one needs it too.
2. **[minor] Redundant field writes in Create.** The follow-up `UpdateUser` call re-sends `Name`/`PhotoUrl`/`Nickname` that were already set in `CreateUser`. Only `ClusterRole` requires the second call.

**Comment by casey-brooks** [comment]
## Summary
- updated agyn_user Create to only send cluster_role in the follow-up UpdateUser call
- added agyn_user acceptance tests for basic create/read, cluster_role update, and import

## Test & Lint Summary
- gofmt -w internal/resources/user_resource.go internal/provider/resource_user_test.go
- go test ./...

**Comment by casey-brooks** [comment]
## Summary
- updated agyn_user Create to only update cluster_role
- added agyn_user acceptance tests for basic create/read, cluster_role update, and import

## Test & Lint Summary
- gofmt -w internal/resources/user_resource.go internal/provider/resource_user_test.go
  - lint: no issues
- go test -json ./... > /tmp/go-test.json
  - passed: 27, failed: 0, skipped: 47

**Review by noa-lucent — DISMISSED** [review]
Both prior items addressed:

1. **Acceptance tests added** — `resource_user_test.go` covers basic create, cluster role update (`none` → `admin`), and import. Follows the established provider test pattern.
2. **Create method cleaned up** — `UpdateUser` now only fires for `ClusterRole`, removing the redundant field writes.

All threads resolved. LGTM — approving.

**Comment by casey-brooks** [comment]
## Summary
- rebased noa/issue-362 onto main and resolved the buf.lock conflict
- re-ran buf generate (no output changes)

## Test & Lint Summary
- buf generate
- go test -json ./... > /tmp/go-test.json
  - passed: 29, failed: 0, skipped: 47

**Comment by casey-brooks** [comment]
## Summary
- rebased noa/issue-362 onto main and resolved the buf.lock conflict
- re-ran buf generate (no output changes)

## Test & Lint Summary
- buf generate
- go test -json ./... > /tmp/go-test.json
  - passed: 29, failed: 0, skipped: 47
- lint: not run (no linter configured)

**Review by noa-lucent — APPROVED** [review]
Re-reviewed after rebase. All code is unchanged from my previous (dismissed) approval — the only diff delta is the removal of `buf.lock` which was already merged to main.

Both prior feedback items remain addressed:
1. Acceptance tests covering basic create, cluster role update, and import.
2. Create method cleanly separated — only calls UpdateUser for ClusterRole.

No new issues. LGTM.

**Review by vitramir — APPROVED** [review]
